### DISCUSSION TITLE
Is there an option to define the timeout for the proxied app?

### DISCUSSION THREAD

**Discussion by dmitry-mightydevops** [gravitational/teleport#34513] [Q&A] [answered]
hi,
I'm running grafana behind teleport 14.1. And sometimes I run a complex loki query over long period of time - like 30 or 90 days which takes 2-5 minutes to execute. In some cases I get my grafana returning no data (around 55s), yet I see the query is active inside loki. Grafana shows no data in the Inspector / stats

![image](https://github.com/gravitational/teleport/assets/70822710/f8178ac0-9f31-42fb-82d4-bbcc3aa49446)

I found nothing in here:
https://goteleport.com/docs/reference/config/#application-service

### any ideas?

**Comment by webvictim** [comment, marked as answer]
There was a fix for this added in 14.2.1 - please update and retry!

https://github.com/gravitational/teleport/pull/34843

**Comment by dmitry-mightydevops** [comment]
yes indeed - this has fixed the issue (assuming your have all correct timeout sets for loki, ingress-nginx of loki gateway and grafana dataproxy, datasource etc)

**Reply by webvictim** [reply]
The PR just removes the timeout entirely so it will wait as long as needed (which is how it worked in the past)
